### PR TITLE
consolidate remaining docs

### LINKED ISSUE #270: Copy/pasted documention in READMEs
Part of the reference guide is copy/pasted in the README. This makes it quite hard to be consistent. The PR I've recently submitted makes the readme of `spring-cloud-starter-config` inconsistent.

### PR DESCRIPTION
**Pull request by dmitry-s**
Fixes #270 and links samples

### CODE REVIEW

File: spring-cloud-gcp-docs/src/main/asciidoc/spring-integration-storage.adoc [hunk 1]
```diff
@@ -20,6 +20,10 @@ Maven coordinates, using Spring Cloud GCP BOM:
     <groupId>org.springframework.cloud</groupId>
     <artifactId>spring-cloud-gcp-pubsub</artifactId>
```
Comment (meltsufin) [inline spring-cloud-gcp-docs/src/main/asciidoc/spring-integration-storage.adoc]: Why is Pub/Sub dependency needed? @joaoandremartins

File: README.adoc [hunk 2]
```diff
@@ -93,8 +93,8 @@ spring.cloud.gcp.credentials.scopes=[SCOPE_1],[SCOPE_2],[SCOPE_3]
 ----
 
 These properties are optional and, if not specified, Spring Boot will attempt to automatically find
-them for you. For details on how Spring Boot finds these properties, refer to the desired starter
-README.
+them for you. For details on how Spring Boot finds these properties, refer to
+the http://cloud.spring.io/spring-cloud-gcp[documentation].
```
Comment (joaoandremartins) [inline README.adoc]: Nit: the link isn't exactly the documentation :P

File: spring-cloud-gcp-docs/src/main/asciidoc/spring-integration-storage.adoc (hunk 1 above)
Comment (joaoandremartins) [reply spring-cloud-gcp-docs/src/main/asciidoc/spring-integration-storage.adoc]: Wow, this is a mistake I discovered a long time ago and I think I fixed on a PR that didn't get to be merged.
Needs to be removed asap!

File: spring-cloud-gcp-docs/src/main/asciidoc/spring-integration-storage.adoc [hunk 3]
```diff
@@ -32,6 +36,7 @@ Gradle coordinates:
 ----
 dependencies {
     compile group: 'org.springframework.cloud', name: 'spring-cloud-gcp-pubsub'
```
Comment (joaoandremartins) [inline spring-cloud-gcp-docs/src/main/asciidoc/spring-integration-storage.adoc]: Remove this too

File: spring-cloud-gcp-docs/src/main/asciidoc/config.adoc [hunk 4]
```diff
@@ -29,6 +29,8 @@ dependencies {
 }
 ----
 
+Sample application is https://github.com/spring-cloud/spring-cloud-gcp/tree/{git_version}/spring-cloud-gcp-samples/spring-cloud-gcp-config-sample[available].
```
Comment (meltsufin) [inline spring-cloud-gcp-docs/src/main/asciidoc/config.adoc]: I would instead put the link on the "sample application".

```
A https://github.com/spring-cloud/spring-cloud-gcp/tree/{git_version}/spring-cloud-gcp-samples/spring-cloud-gcp-config-sample[sample application] that uses this feature is available.
```

File: spring-cloud-gcp-docs/src/main/asciidoc/getting-started.adoc [hunk 5]
```diff
@@ -42,7 +42,8 @@ The GCP Storage entry adds the GCP Support entry and all the required dependenci
 
 === Code Samples
 
-There are https://github.com/spring-cloud/spring-cloud-gcp/tree/master/spring-cloud-gcp-samples[code samples] available that demonstrate the usage of all our integrations.
+There are https://github.com/spring-cloud/spring-cloud-gcp/tree/{git_version}/spring-cloud-gcp-samples[code samples] available that demonstrate the usage of all our integrations.
+https://github.com/spring-cloud/spring-cloud-gcp/tree/{git_version}/spring-cloud-gcp-samples/spring-cloud-gcp-vision-api-sample[This one] shows how to use `spring-cloud-gcp-starter` for authentication.
```
Comment (meltsufin) [inline spring-cloud-gcp-docs/src/main/asciidoc/getting-started.adoc]: "This one" -> "The Vision API sample"

File: spring-cloud-gcp-docs/src/main/asciidoc/logging.adoc [hunk 6]
```diff
@@ -31,6 +31,8 @@ NOTE: Due to the way logging is set up, the GCP project ID and credentials defin
 Instead, you should set the `GOOGLE_CLOUD_PROJECT` and `GOOGLE_APPLICATION_CREDENTIALS` environment variables to the project ID and credentials private key location, respectively.
 You can do this easily if you're using the http://cloud.google.com/sdk[Google Cloud SDK], using the `gcloud config set project [YOUR_PROJECT_ID]` and `gcloud auth application-default login` commands, respectively.
 
+Sample application is https://github.com/spring-cloud/spring-cloud-gcp/tree/{git_version}/spring-cloud-gcp-samples/spring-cloud-gcp-logging-sample[available].
```
Comment (meltsufin) [inline spring-cloud-gcp-docs/src/main/asciidoc/logging.adoc]: Again, something like:

```
A https://github.com/spring-cloud/spring-cloud-gcp/tree/{git_version}/spring-cloud-gcp-samples/spring-cloud-gcp-logging-sample[Logging sample application] is available.
```

File: spring-cloud-gcp-docs/src/main/asciidoc/sql.adoc [hunk 7]
```diff
@@ -47,6 +47,12 @@ for "Cloud SQL API", click the first result and enable the API.
 NOTE: There are several similar "Cloud SQL" results. You must access the "Google Cloud SQL API" one
 and enable the API from there.
 
+Sample applications:
```
Comment (meltsufin) [inline spring-cloud-gcp-docs/src/main/asciidoc/sql.adoc]: "Available sample applications:"

**Pull request merged**